plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -321,6 +329,14 @@ runCompiler opts expr = do
     -- GHC.Core -> Pir translation.
     pirT <- PIR.runDefT () $ compileExprWithDefs expr
 
+    -- dumping binary pir
+    -- NOTE:  consider doing it after simplification
```

Comment:
@michaelpj yes, could do that.

Speaking about names, do you know how to thread the name of the module so that I could name the file `pir-module-name.flat`?  I am not even sure if the plugin even holds such information.

Reply:
Let's make the dumping consistent. I'm sure we can get the module name, but don't worry about it for now.

Reply:
@michaelpj, regarding the simplified version, there does not seem to be a flat instance for `PIR.Provenance`, I think you can revisit dumping simplified pir terms once it is added---I didn't simply add a `Generic` instance for `PIR.Provenance`, because it contains a case with a `Set`, and I am not sure if there is a `Flat` instance of a `Set`.